controllers/articles-controller.js

```diff
@@ -11,5 +11,20 @@ function getEndpoints(req,res,next){
   res.status(200).send({ endpoints })
 }
 
+function getArticleById(req,res,next){
+    const { article_id } = req.params
+    selectArticleById(article_id)
+    .then((article) => {
+      if(!article){
+        res.status(404).send({ msg: 'article does not exist' })
+      }
```

Comment:
The model should be in charge of all data manipulation. This should be caught in the model, preferably using Promise.reject to send the error. Doing it this way may need refactoring in the future 👍 keeping your controller clean and strictly just sending data is preferred.